src/common/k8s.ts

```diff
@@ -371,15 +371,23 @@ export async function patchContainerResourcesOfSts(
     for (const pod of pods.items) {
       const actualResources = pod.spec?.containers?.find((container) => container.name === containerName)?.resources
 
-      if (actualResources != desiredResources) {
-        d.info(`sts/argocd-application-controller pod has not desired resources`)
+      if (
+        isEqual(actualResources?.limits, desiredResources?.limits) &&
+        isEqual(actualResources?.requests, desiredResources?.requests)
+      ) {
+        d.info(
+          `sts/argocd-application-controller pod has desired resources: ${JSON.stringify(
+            desiredResources,
+          )} and actual resources: ${JSON.stringify(actualResources)}`,
+        )
+        return
+      }
 
-        await patchStatefulSetResources(statefulSetName, containerName, namespace, desiredResources, appsApi, d)
-        d.info(`sts/argocd-application-controller has been patched with resources: ${JSON.stringify(desiredResources)}`)
+      await patchStatefulSetResources(statefulSetName, containerName, namespace, desiredResources, appsApi, d)
+      d.info(`sts/argocd-application-controller has been patched with resources: ${JSON.stringify(desiredResources)}`)
 
-        await deleteStatefulSetPods(statefulSetName, namespace, appsApi, coreApi, d)
-        d.info(`sts/argocd-application-controller pods restarted`)
-      }
+      await deleteStatefulSetPods(statefulSetName, namespace, appsApi, coreApi, d)
```

Comment:
why deleteing Pods? Is sts not going to take care of it?

Reply:
we want to ensure that this change is applied immediately, restarting argocd pods do not seem to be problematic